### ISSUE TITLE
Drop dependency on insert-ordered-containers

### ISSUE THREAD

**Issue by jwiegley** [haskell-nix/hnix#161]
@mightybyte Pulling in `insert-ordered-containers` causes us to transitively depend on `lens`. Without using this package, we would otherwise have a pretty small dependency set nowadays (after the switch to `megaparsec`).

Can we just copy out of `insert-ordered-containers` the bit that we need?

**Comment by jwiegley** [comment]
Since we only use this for parameter sets, and those generally are smallish in scale, I'm going to replace it with a regular association list for now.

**Comment by jwiegley** [comment]
Here's our current transitive dependency set:

```
Glob-0.9.2
QuickCheck-2.10.1
StateVar-1.1.0.4
adjunctions-4.4
aeson-1.2.4.0
ansi-terminal-0.8.0.2
ansi-wl-pprint-0.6.8.2
asn1-encoding-0.9.5
asn1-parse-0.9.4
asn1-types-0.3.2
async-2.1.1.1
attoparsec-0.13.2.2
base-compat-0.9.3
base-orphans-0.7
base16-bytestring-0.1.1.6
basement-0.0.7
bifunctors-5.5.2
blaze-builder-0.4.1.0
boehm-gc-7.6.4-dev
byteable-0.1.1
byteable-0.1.1-data
call-stack-0.1.0
case-insensitive-1.2.0.11
cereal-0.5.5.0
clock-0.7.2
colour-2.3.4
colour-2.3.4-data
comonad-5.0.3
conduit-1.3.0.2
connection-0.2.8
contravariant-1.4.1
cookie-0.4.4
cpphs-1.20.8
cryptohash-0.11.9
cryptonite-0.25
data-default-class-0.1.2.0
data-fix-0.2.0
deriving-compat-0.4.1
distributive-0.5.3
dlist-0.8.0.4
exceptions-0.8.3
fail-4.9.0.0
foundation-0.0.20
free-5.0.1
ghc-8.2.2
gmp-6.1.2-dev
hashable-1.2.7.0
haskell-src-exts-1.20.2
haskell-src-meta-0.8.0.2
hourglass-0.2.11
hpack-0.27.0
hscolour-1.24.4
http-client-0.5.11
http-client-tls-0.3.5.3
http-types-0.12.1
integer-logarithms-1.0.2.1
interpolate-0.2.0
kan-extensions-5.1
keys-3.12
libyaml-0.1.7
megaparsec-6.4.1
memory-0.14.16
mime-types-0.1.0.7
monadlist-0.0.2
mono-traversable-1.0.8.1
mtl-2.2.2
network-2.6.3.4
network-uri-2.6.1.0
nix-2.0-dev
old-locale-1.0.0.7
old-time-1.1.0.3
optparse-applicative-0.14.2.0
parsec-3.1.13.0
parser-combinators-0.4.0
pem-0.2.4
pointed-5.0.1
polyparse-1.12
primitive-0.6.3.0
profunctors-5.2.2
random-1.1
regex-base-0.93.2
regex-tdfa-1.2.3
regex-tdfa-text-1.0.0.3
resourcet-1.2.0
safe-0.3.17
scientific-0.3.5.3
semigroupoids-5.2.2
semigroups-0.18.4
socks-0.5.6
split-0.2.3.3
stdenv-linux
stm-2.4.5.0
streaming-commons-0.1.19
syb-0.7
tagged-0.8.5
tasty-1.0.1.1
tasty-hunit-0.10.0.1
tasty-th-0.1.7
text-1.2.3.0
tf-random-0.5
th-abstraction-0.2.6.0
th-expand-syns-0.4.4.0
th-lift-0.7.8
th-lift-instances-0.1.11
th-orphans-0.13.5
th-reify-many-0.1.8
these-0.7.4
time-locale-compat-0.1.1.3
tls-1.4.1
transformers-base-0.4.4
transformers-compat-0.5.1.4
unbounded-delays-0.1.1.0
unliftio-core-0.1.1.0
unordered-containers-0.2.9.0
uuid-types-1.0.3
vector-0.12.0.1
vector-algorithms-0.7.0.1
vector-instances-3.4
void-0.7.2
x509-1.7.3
x509-store-1.6.6
x509-system-1.6.6
x509-validation-1.6.10
xml-1.3.14
yaml-0.8.28
zlib-0.6.2
zlib-1.2.11-dev
```
